app/src/main/java/io/homeassistant/companion/android/sensors/BluetoothSensorManager.kt

```diff
@@ -60,35 +76,67 @@ class BluetoothSensorManager : SensorManager {
-                "paired_devices" to bondedString
-            )
+                context,
+                bluetoothConnection,
+                totalConnectedDevices,
+                icon,
+                mapOf(
+                        "connected_paired_devices" to connectedPairedDevices,
+                        "connected_not_paired_devices" to connectedNotPairedDevices,
+                        "paired_devices" to bondedString
+                )
         )
     }
 
+    private fun isBtOn(context: Context): Boolean {
+        var btOn = false
+        if (checkPermission(context, bluetoothState.id)) {
+            btOn = BluetoothUtils.isOn(context)
+        }
+        return btOn
+    }
+
     private fun updateBluetoothState(context: Context) {
         if (!isEnabled(context, bluetoothState.id))
             return
+        val icon = if (isBtOn(context)) "mdi:bluetooth" else "mdi:bluetooth-off"
+        onSensorUpdated(
+                context,
+                bluetoothState,
+                isBtOn(context),
+                icon,
+                mapOf()
+        )
+    }
 
-        var isBtOn = false
-
-        if (checkPermission(context, bluetoothState.id)) {
-            isBtOn = BluetoothUtils.isOn(context)
+    private fun updatedBLEDevice(context: Context): Boolean {
+        var result = false
+        var id1 = getSetting(context, bleTransmitter, BLE_ID1, "string", DEFAULT_BLE_ID1)
+        var id2 = getSetting(context, bleTransmitter, BLE_ID2, "string", DEFAULT_BLE_ID2)
+        var id3 = getSetting(context, bleTransmitter, BLE_ID3, "string", DEFAULT_BLE_ID3)
+        if (bleTransmitterDevice.id1 != id1 || bleTransmitterDevice.id2 != id2 || bleTransmitterDevice.id3 != id3) {
+            result = true
+            bleTransmitterDevice.id1 = id1
+            bleTransmitterDevice.id2 = id2
+            bleTransmitterDevice.id3 = id3
         }
-        val icon = if (isBtOn) "mdi:bluetooth" else "mdi:bluetooth-off"
+        return result
+    }
 
+    private fun updateBLEtrasnmitter(context: Context) {
+        if (!isEnabled(context, bleTransmitter.id) && bleTransmitterDevice.transmitting) // sensor has been turned off, stop transmitting
+            TransmitterManager.stopTransmitting(bleTransmitterDevice)
+        else if ((isEnabled(context, bleTransmitter.id) && !bleTransmitterDevice.transmitting) || updatedBLEDevice(context)) // sensor is on, start transmitting (which only start is not already running, or details have changed
+            TransmitterManager.startTransmitting(context, bleTransmitterDevice)
+        val icon = if (bleTransmitterDevice.transmitting) "mdi:bluetooth" else "mdi:bluetooth-off"
         onSensorUpdated(
-            context,
-            bluetoothState,
-            isBtOn,
-            icon,
-            mapOf()
+                context,
+                bleTransmitter,
+                bleTransmitterDevice.state,
```

Comment:
There appears to be 3 different states that can be sent? how does that work for this `binary_sensor`?  or is this supposed to be a normal `sensor` :)

Reply:
At the moment there are 3 states, Transmitting, ErrorTransmitting and Stopped. Stoppped should only happened if BT is disabled wheb we try and transmit.

I'll change stopped state words to indicate BT not enabled.

In future merge I'd like to add a listener to BT adapter turning on and off, to maintain transmitter state and start when re-enabled.

Reply:
Ok if we wish to remain as a `binary_sensor` then we should report `true` for the state if we are `transmitting` only, otherwise if you want the state to say `transmitting` and the others we will need to change the sensor type to `sensor`

https://github.com/home-assistant/android/blob/4300a9bc5ec3a2c86fc9ca2520e74d34a27bee29/app/src/main/java/io/homeassistant/companion/android/sensors/BluetoothSensorManager.kt#L35

Reply:
Gah, sorry, missed that, will change to sensor.